modules/map/tools/map_datachecker/server/channel_verify.cc

```diff
@@ -0,0 +1,228 @@
+}
+
+ErrorCode ChannelVerify::check(std::string record_dir_or_record_full_path) {
+  std::vector<std::string> records_path =
+    get_records_path(record_dir_or_record_full_path);
+  if (records_path.size() == 0) {
+    AINFO << "have no data file to check";
+    _return_state = ErrorCode::ERROR_VERIFY_NO_RECORDERS;
+    return _return_state;
+  }
+  incremental_check(records_path);
+  _return_state = ErrorCode::SUCCESS;
+  return _return_state;
+}
+
+std::shared_ptr<std::vector<OneRecordChannelCheckResult>>
+  ChannelVerify::get_check_result() {
+  return _sp_vec_check_result;
+}
+
+int ChannelVerify::incremental_check(std::vector<std::string> records_path) {
+  // 1.
+  std::vector<std::string> not_check_records_path;
+  AINFO << "all records path:";
+  for (size_t i = 0; i < records_path.size(); ++i) {
+    AINFO << "[" << i << "]: " << records_path[i];
+    if (!is_record_checked(records_path[i])) {
+      not_check_records_path.push_back(records_path[i]);
+    }
+  }
+
+  // 2.
+  AINFO << "not_check_records_path:";
+  for (size_t i = 0; i < not_check_records_path.size(); ++i) {
+    AINFO << "[" << i << "]: " << not_check_records_path[i];
+    OneRecordChannelCheckResult check_result =
+      check_record_channels(not_check_records_path[i]);
+    _sp_vec_check_result->push_back(check_result);
+  }
+
+  return 0;
+}
+
+bool ChannelVerify::is_record_file(std::string record_path) {
+  if (!boost::filesystem::exists(record_path)) {
+    AINFO << "path [" << record_path << "] is not exist";
+    return false;
+  }
+  if (!boost::filesystem::is_regular_file(record_path)) {
+    AINFO << "path [" << record_path << "] is not a regular file";
+    return false;
+  }
+  if (boost::filesystem::extension(record_path) != ".record") {
+    AINFO << "path [" << record_path << "] extension is not record";
+    return false;
+  }
+  // TODO(yuanyijun): check if record_path is a real record file
+  // To avoid disk overhead caused by opening files twice, the real
+  // file checking is placed in the function [ChannelVerify::get_record_info]
+  return true;
+}
+
+std::vector<std::string> ChannelVerify::get_records_path(
+  std::string record_dir_or_record_full_path) {
+  // record_dir_or_record_full_path is record fullpath or
+  // directory which contains some records
+  std::vector<std::string> records_path;
+  // 1. check record_dir_or_record_full_path is valid or not
+  boost::filesystem::path path(record_dir_or_record_full_path);
+  if (!boost::filesystem::exists(path)) {
+    AINFO << "record path ["
+          << record_dir_or_record_full_path
+          << "] is not exist";
+    return records_path;
+  }
+
+  if (is_record_file(record_dir_or_record_full_path)) {
+    records_path.push_back(record_dir_or_record_full_path);
+  } else if (boost::filesystem::is_directory(path)) {
+    using dit_t = boost::filesystem::directory_iterator;
+    dit_t end;
+    for (dit_t it(record_dir_or_record_full_path); it != end; ++it) {
```

Comment:
You can use auto.

Reply:
There's a problem with using auto in this place.